src/util/invariant.cpp

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************\
+
+Module:
+
+Author: Martin Brain, martin.brain@diffblue.com
+
+\*******************************************************************/
+
+
+#include "invariant.h"
+
+#include <stdexcept>
+#include <string>
+#include <sstream>
+
+
+#ifdef CPROVER_INVARIANT_PRINT_STACK_TRACE
+#include <iostream>
+#include <stdlib.h>
+#endif
+
+// Backtraces compiler and C library specific
+#include <assert.h>
```

Comment:
That won't do what I'm after.  I'm including a C header so that I get __GLIBC__ defined if that C library is GLibc.  Thus I want something that is definitely and always a C header, regardless of how the stdc++ library is implemented.